integ-tests/src/cucumber/java/steps/dataservice/DescribeEnvironmentSteps.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package steps.dataservice;
+
+import com.amazonaws.blox.dataservicemodel.v1.model.Environment;
+import com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentId;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.CreateEnvironmentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.CreateEnvironmentResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.DescribeEnvironmentResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.UpdateEnvironmentResponse;
+
+import configuration.CucumberConfiguration;
+import cucumber.api.java8.En;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import steps.helpers.InputCreator;
+import steps.wrappers.DataServiceWrapper;
+import static org.junit.Assert.assertTrue;
+
+@ContextConfiguration(classes = CucumberConfiguration.class)
+public class DescribeEnvironmentSteps implements En {
+
+  @Autowired private DataServiceWrapper dataServiceWrapper;
+  @Autowired private InputCreator inputCreator;
+
+  public DescribeEnvironmentSteps() {
+
+    When(
+        "^I describe the created environment$",
+        () -> {
+          final EnvironmentId environmentId = getEnvironmentIdFromCreatedEnvironment();
+          dataServiceWrapper.describeEnvironment(
+              inputCreator.describeEnvironmentRequest(
+                  environmentId.getEnvironmentName(), environmentId.getCluster()));
+        });
+
+    Then(
+        "^the created and described environments match$",
+        () -> {
+          final CreateEnvironmentResponse createEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(CreateEnvironmentResponse.class);
+          final DescribeEnvironmentResponse describeEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(DescribeEnvironmentResponse.class);
+          checkEnvironmentEquality(
+              createEnvironmentResponse.getEnvironment(),
+              describeEnvironmentResponse.getEnvironment());
+        });
+
+    Given(
+        "^I update the created environment$",
+        (final String environmentNamePrefix) -> {
+          final EnvironmentId environmentId = getEnvironmentIdFromCreatedEnvironment();
+          dataServiceWrapper.updateEnvironment(
+              inputCreator.updateEnvironmentRequest(
```

Comment:
doesn't this need to contain a field that you're updating? it might make sense to pass in an environment object so you don't have to add all possible updatable fields to the method signature

Reply:
Looks like I can't pass the whole environment object in? The parameter of the step definition need to be the same as the one in the feature definitions, which can only be string/numbers. 

I can modify this step into "I update the created environment with new task definition "new task definition" "